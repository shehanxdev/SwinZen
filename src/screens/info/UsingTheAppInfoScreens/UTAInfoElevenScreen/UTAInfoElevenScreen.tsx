import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { TextAlignment, TextVariant } from '@sz/constants';

import { BaseInfoScreen } from '../../components';

export function UTAInfoElevenScreen() {
  return (
    <BaseInfoScreen>
      <View style={tw`px-4 py-6`} testID="UTAInfoElevenScreenTestID">
        <Text variant={TextVariant.Body2Regular} textAlign={TextAlignment.Auto}>
          UTAInfoElevenScreen
        </Text>
      </View>
    </BaseInfoScreen>
  );
}
