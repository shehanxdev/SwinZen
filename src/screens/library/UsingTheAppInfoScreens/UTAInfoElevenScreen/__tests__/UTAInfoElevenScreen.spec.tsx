import { render } from '@testing-library/react-native';
import * as React from 'react';

import { UTAInfoElevenScreen } from '../UTAInfoElevenScreen';

describe('UTAInfoEleven Screen', () => {
  const testID = 'UTAInfoElevenScreenTestID';

  const getRenderedScreen = () => render(<UTAInfoElevenScreen />);

  it(`should render UTAInfoEleven Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the UTAInfoEleven Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
