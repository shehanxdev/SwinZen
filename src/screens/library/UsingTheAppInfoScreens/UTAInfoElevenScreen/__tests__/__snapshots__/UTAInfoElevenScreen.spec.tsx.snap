// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UTAInfoEleven Screen should render UTAInfoEleven Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 32,
              }
            }
            testID="UTAInfoElevenScreenTestID"
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              SwingZen scores pass/fail and help you track your progress.
          
If you fail a certain checkpoint, you can progress to view the instructions on how to improve that portion of the swing. If your position is good, then great! If not, you will then have the information through AI Pro Tips, Pro vs you, PGA instruction videos, and drills to correct that specific swing fault.
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
